lib/server.js

```diff
@@ -41,7 +41,7 @@ function createServer (options, httpHandler) {
       listenOptions.cb = cb
     }
 
-    const { host = 'localhost' } = listenOptions
+    const { host = listenOptions.path == null ? 'localhost' : null } = listenOptions
```

Comment:
```suggestion
    // <insert link to GH issue here>
    const { host = listenOptions.path == null ? 'localhost' : null } = listenOptions
```

This is a little difficult to reason through. The one line change is clever, but I think we should expand it out into a full `if` conditional with a comment or two to explain why this is being done.

Reply:
Let me know if you think there is some clearer way to write this.